core/src/main/clojure/xtdb/node/impl.clj

```diff
@@ -111,8 +113,13 @@
 
   (open-sql-query [this query query-opts]
     (let [query-opts (-> query-opts (with-query-opts-defaults this))]
-      (-> (xtp/prepare-sql this query query-opts)
-          (then-execute-prepared-query allocator query-timer query-opts))))
+      (try
+        (-> (xtp/prepare-sql this query query-opts)
+            (then-execute-prepared-query allocator query-timer query-opts))
+        (catch Exception e
+          (when query-error-counter
+            (.increment ^Counter query-error-counter))
```

Comment:
I think you are double counting here. If you follow `then-execute-prepared-query` you end up in `wrap-query` at some point. With all the different phases this might be a bit hard to see, I agree. Maybe let's pair on this, so we can best figure out where to put the counters on the node and in pgwire.

Reply:
So lets add the `error-counter` on the `PLocalNode` methods. `prepare-sql`, `prepare-sql` and `prepare-ra`.